### ISSUE TITLE
`dotnet restore` documentation is out of date

### ISSUE THREAD

**Issue by dtivel** [dotnet/docs#11936]
**Before you open an issue**

If the issue is:

- A simple typo or similar correction, you can submit a PR. See [the contributor guide](https://docs.microsoft.com/contribute/#quick-edits-to-existing-documents) for instructions.
- A general support question, consider asking on a support forum site.
- A site design concern, create an issue at [MicrosoftDocs/feedback](https://github.com/MicrosoftDocs/feedback/issues/new/choose).
- A problem completing a tutorial, compare your code with the completed sample.
- A duplicate of an open or closed issue, leave a comment on that issue.

**Issue description**

https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-restore.md is not current.

Execute:  `dotnet restore -?`
Output:
```
Usage: dotnet restore [options] <PROJECT>

Arguments:
  <PROJECT>   The project file to operate on. If a file is not specified, the command will search the current directory for one.

Options:
  -h, --help                           Show command line help.
  -s, --source <SOURCE>                The NuGet package source to use for the restore.
  -r, --runtime <RUNTIME_IDENTIFIER>   The target runtime to restore packages for.
  --packages <PACKAGES_DIR>            The directory to restore packages to.
  --disable-parallel                   Prevent restoring multiple projects in parallel.
  --configfile <FILE>                  The NuGet configuration file to use.
  --no-cache                           Do not cache packages and http requests.
  --ignore-failed-sources              Treat package source failures as warnings.
  --no-dependencies                    Do not restore project-to-project references and only restore the specified project.
  -f, --force                          Force all dependencies to be resolved even if the last restore was successful.
                                       This is equivalent to deleting project.assets.json.
  -v, --verbosity <LEVEL>              Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
  --interactive                        Allows the command to stop and wait for user input or action (for example to complete authentication).
  --use-lock-file                      Enables project lock file to be generated and used with restore.
  --locked-mode                        Don't allow updating project lock file.
  --lock-file-path <LOCK_FILE_PATH>    Output location where project lock file is written. By default, this is 'PROJECT_ROOT\packages.lock.json'.
  --force-evaluate                     Forces restore to reevaluate all dependencies even if a lock file already exists.
```
Issues:
* The `-f` shortcut is not mentioned in documentation.
* The `-v` shortcut is not mentioned in documentation.
* The `--use-lock-file` option is not mentioned in documentation.
* The `--locked-mode` option is not mentioned in documentation.
* The `--lock-file-path` option is not mentioned in documentation.
* The `--force-evaluate` option is not mentioned in documentation.

**Target framework**

Check the .NET target framework(s) being used, and include the version number(s).

- [X] .NET Core
- [ ] .NET Framework
- [ ] .NET Standard

If using the .NET Core SDK, include `dotnet --info` output. If using .NET Framework without the .NET Core SDK, include info from Visual Studio's **Help** > **About Microsoft Visual Studio** dialog.

<details>
<summary><strong>dotnet --info output</strong> or <strong>About VS info</strong></summary>

```console
.NET Core SDK (reflecting any global.json):
 Version:   2.2.106
 Commit:    aa79b139a8

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\dtivel\Documents\dnc\sdk\2.2.106\

Host (useful for support):
  Version: 2.2.4
  Commit:  f95848e524

.NET Core SDKs installed:
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.700-preview-009597 [C:\Program Files\dotnet\sdk]
  2.2.106 [C:\Users\dtivel\Documents\dnc\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.4 [C:\Users\dtivel\Documents\dnc\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.4 [C:\Users\dtivel\Documents\dnc\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.4 [C:\Users\dtivel\Documents\dnc\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```
</details>

edit by @mairaw: added article details

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0b256dc4-c4ad-c115-f4e2-40310bc3505a
* Version Independent ID: 0b993779-65a2-6092-4047-2fc2ec16170b
* Content: [dotnet restore command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-restore?tabs=netcore2x)
* Content Source: [docs/core/tools/dotnet-restore.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-restore.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

**Comment by mairaw** [comment]
Thanks for your feedback @dtivel. You're correct. We are in the process of updating the CLI docs for 2.2+ and this update is scheduled for our next sprint.

**Comment by mairaw** [comment]
I'm moving this item to the backlog and will pick it up once I start the work to update this article for the 3.0 release as well. It should happen soon, hopefully.

**Comment by minuz** [comment]
Hi @Mairaw, can you please add reference to the credentials plugin? 
https://github.com/microsoft/artifacts-credprovider

I use a private nuget feed and the default dotnet restore does not prompt the user to add the authentication to the private feed and you end up not being able to restore the project/solution.
It took me a while to find the existence of the plugin. It would be nice to have it on the official documentation.

**Comment by minuz** [comment]
Ops, forgot to mention that I found this issue on MacOS. The windows version actually find the credential if you are using active directory and the private feed uses the same credential.

**Comment by mairaw** [comment]
@KathleenDollard @dsplaisted @nguerrera do you know when some of the new options for dotnet restore were added to the CLI? Is there a good way to find this?

**Comment by nguerrera** [comment]
@nkolev92

**Comment by nkolev92** [comment]
* -f 
* -v

The above 2 have been around since the 1.x days.

* --use-lock-file
* --locked-mod
* --lock-file-path
* --force-evaluate

These ones have been around since 4.9.x of NuGet, see release notes: https://docs.microsoft.com/en-us/nuget/release-notes/nuget-4.9-rtm

and that makes it 2.1.500 and 2.2.100 of the SDK.
